pkg/response/formatter.go

```diff
@@ -0,0 +1,211 @@
+package response
+
+import (
+	"bytes"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"gopkg.in/yaml.v3"
+
+	"github.com/leaktk/scanner/pkg/config"
+	"github.com/leaktk/scanner/pkg/logger"
+)
+
+// OutputFormat is the code(int) for each format
+type OutputFormat int
+
+const (
+	// JSON displays the output in JSON format
+	JSON OutputFormat = iota
+	// HUMAN displays the outut in a way that's nice for humans to read
+	HUMAN
+	// TOML displays the output in TOML format
+	TOML
+	// YAML displays the output in YAML format
+	YAML
+	// CSV displays the
+	CSV
+)
+
+type Formatter struct {
+	format OutputFormat
+}
+
+// NewFormatter creates new formatter
+func NewFormatter(cfg config.Formatter) (*Formatter, error) {
+	format, err := GetOutputFormat(cfg.Format)
+	if err != nil {
+		return nil, err
+	}
+	return &Formatter{format: format}, nil
+}
+
+// GetOutputFormat takes the string and returns OutputFormat or an error
+func GetOutputFormat(format string) (OutputFormat, error) {
+	format = strings.ToUpper(format)
+	switch format {
+	case "JSON":
+		return JSON, nil
+	case "HUMAN":
+		return HUMAN, nil
+	case "TOML":
+		return TOML, nil
+	case "YAML":
+		return YAML, nil
+	case "CSV":
+		return CSV, nil
+	default:
+		return JSON, fmt.Errorf("invalid output format option: format=%q", format)
+	}
+}
+
+// SetFormat sets the format using the OutputFormat value
+func (f *Formatter) SetFormat(format OutputFormat) {
+	f.format = format
+}
```

Comment:
```suggestion
```

This can be removed, correct?

It doesn't seem to be used anywhere and it doesn't seem like we're mutating the formatter after it's created.